pkg/fleet/installer/service/docker.go

```diff
@@ -117,6 +127,67 @@ func (a *apmInjectorInstaller) deleteDockerConfigContent(previousContent []byte)
 	return dockerConfigJSON, nil
 }
 
+// verifyDockerConfig validates the docker daemon configuration for latest
+// docker versions (>=23.0.0)
+// For docker versions <23.0.0, the validation is skipped but we rely on the post-reload
+// verification to check if the configuration has been applied properly
+func (a *apmInjectorInstaller) verifyDockerConfig(path string) error {
+	cmd := exec.Command("docker", "version", "-f", "{{.Client.Version}}")
+	versionBuffer := new(bytes.Buffer)
+	cmd.Stdout = versionBuffer
+	err := cmd.Run()
+	if err != nil {
+		log.Warnf("failed to get docker version: %s, skipping verification", err)
+		return nil
+	}
+	dockerVersionSplit := strings.Split(versionBuffer.String(), ".")
+	if len(dockerVersionSplit) < 3 {
+		log.Warnf("failed to parse docker version %s, skipping verification", versionBuffer.String())
+	}
+	majorDockerVersionInt, err := strconv.Atoi(dockerVersionSplit[0])
+	if err != nil {
+		log.Warnf("failed to parse docker version %s: %s, skipping verification", dockerVersionSplit[0], err)
+		return nil
+	}
+	if majorDockerVersionInt < 23 {
+		log.Warnf("docker version %d is not supported for verification, skipping", majorDockerVersionInt)
+		return nil
+	}
+
+	cmd = exec.Command("dockerd", "--validate", "--config-file", path)
+	buf := new(bytes.Buffer)
+	cmd.Stderr = buf
+	err = cmd.Run()
+	if err != nil {
+		return fmt.Errorf("failed to validate docker config (%v): %s", err, buf.String())
+	}
+	return nil
+}
+
+// verifyDockerRuntime validates that docker runtime configuration contains
+// a path to the injector runtime. This must be done because we can't always
+// verify the configuration befor a reload depending on the docker version
+// As the reload is eventually consistent we have to retry a few times
+func (a *apmInjectorInstaller) verifyDockerRuntime() (err error) {
+	for i := 0; i < 3; i++ {
+		if i > 0 {
+			time.Sleep(time.Second)
+		}
+		cmd := exec.Command("docker", "system", "info", "--format", "{{ .DefaultRuntime }}")
+		var outb bytes.Buffer
+		cmd.Stdout = &outb
+		err = cmd.Run()
```

Comment:
Why are you not checking the error under here 🤔

Reply:
We're retrying it and only checking the last one. I will add a log.